src/Plutarch/Test/QuickCheck/Function.hs

```diff
@@ -160,3 +170,24 @@ plamFinite f = plam $ \x -> pfromJust #$ plookup # x # table
   where
     table :: Term s (PBuiltinList (PBuiltinPair a b))
     table = pconstant $ (id &&& f) <$> universeF
+
+pfind' ::
+    forall (a :: S -> Type) (s :: S) list.
+    PIsListLike list a =>
+    (Term s a -> Term s PBool) ->
+    Term s (list a :--> PMaybe a)
+pfind' p =
+    precList
+        (\self x xs -> pif (p x) (pcon (PJust x)) (self # xs))
+        (const $ pcon PNothing)
```

Comment:
I think this [already exists](https://hoogle.nix.dance/file/nix/store/dz4rwcm8cj8wgfmynfaq7gg7ff4sppcr-plutarch-lib-plutarch-1.2.0-haddock-doc/share/doc/plutarch/html/Plutarch-List.html#v:pfind).

Reply:
Great! I should use one form Plutarch itself